service/service.go

```diff
@@ -19,22 +19,106 @@ package service
 import (
 	"context"
 	"errors"
+	"fmt"
+	"time"
 
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
 	pb "github.com/kubeflow/metadata/api"
 	mlmd "github.com/kubeflow/metadata/ml_metadata"
+	"github.com/kubeflow/metadata/schemaparser"
+)
+
+const (
+	wholeMetaPropertyName string = "__ALL_META__"
+	categoryPropertyName  string = "category"
+	artifactCategory      string = "artifact"
+	defaultTimeout               = 5 * time.Second
 )
 
 // Service implements the gRPC service MetadataService defined in the metadata
 // API spec.
 type Service struct {
-	mlmdClient mlmd.MetadataStoreServiceClient
+	mlmdClient       mlmd.MetadataStoreServiceClient
+	typeNameToMLMDID map[string]int64
+
+	// TODO: Once MLMD type system supports schema annotations, the following two fields can be obtained by quering MLMD types.
+	typeNameToID map[string]string
```

Comment:
Why do we need this id btw?

Reply:
Added a comment. this map the incoming type name to the id of corresponding schema and we can verify the incoming data with the schema.